astropy/modeling/core.py

```diff
@@ -678,6 +678,57 @@ def __init__(self, *args, meta=None, name=None, **kwargs):
         # Parameter values must be passed in as keyword arguments in order to
         # distinguish them
         self._initialize_parameters(args, kwargs)
+        self._initialize_unit_support()
+
+    def _initialize_unit_support(self):
+        """
+        Convert self._input_units_strict and
+        self.input_units_allow_dimensionless to dictionaries
+        mapping input name to a boolena value.
+        """
+        if isinstance(self._input_units_strict, bool):
+            self._input_units_strict = {key: self._input_units_strict for
+                                        key in self.__class__.inputs}
+
+        if isinstance(self._input_units_allow_dimensionless, bool):
+            self._input_units_allow_dimensionless = {key: self._input_units_allow_dimensionless
+                                                     for key in self.__class__.inputs}
+
+    @property
+    def input_units_strict(self):
+        """
+        Enforce strict units on inputs to evaluate. If this is set to True, input
+        values to evaluate have to be in the exact right units specified by
```

Comment:
I understand this as if the example below should fail:
```
g  = Gaussian1D(10*u.Hz, 3*u.m, 2*u.m)
g.input_units_strict = True
g(1 * u.cm)
```
It actually works and converts `cm` to `m`.  Am I misunderstanding the docstrings?